crates/nu-command/src/filesystem/rm.rs

```diff
@@ -341,132 +339,136 @@ fn rm(
+                } else if verbose {
+                    let msg = if interactive && !confirmed {
+                        "not deleted"
                     } else {
-                        Value::nothing(span)
-                    }
+                        "deleted"
+                    };
+                    Ok(Some(format!("{} {:}", msg, f.to_string_lossy())))
                 } else {
-                    let error = format!("Cannot remove {:}. try --recursive", f.to_string_lossy());
-                    Value::error(
-                        ShellError::GenericError {
-                            error,
-                            msg: "cannot remove non-empty directory".into(),
-                            span: Some(span),
-                            help: None,
-                            inner: vec![],
-                        },
-                        span,
-                    )
+                    Ok(None)
                 }
             } else {
-                let error = format!("no such file or directory: {:}", f.to_string_lossy());
-                Value::error(
-                    ShellError::GenericError {
-                        error,
-                        msg: "no such file or directory".into(),
-                        span: Some(span),
-                        help: None,
-                        inner: vec![],
-                    },
-                    span,
-                )
+                let error = format!("Cannot remove {:}. try --recursive", f.to_string_lossy());
+                Err(ShellError::GenericError {
+                    error,
+                    msg: "cannot remove non-empty directory".into(),
+                    span: Some(span),
+                    help: None,
+                    inner: vec![],
+                })
+            }
+        } else {
+            let error = format!("no such file or directory: {:}", f.to_string_lossy());
+            Err(ShellError::GenericError {
+                error,
+                msg: "no such file or directory".into(),
+                span: Some(span),
+                help: None,
+                inner: vec![],
+            })
+        }
+    });
+
+    for result in iter {
+        if nu_utils::ctrl_c::was_pressed(&engine_state.ctrlc) {
+            return Err(ShellError::InterruptedByUser {
+                span: Some(call.head),
+            });
+        }
+
+        match result {
+            Ok(None) => {}
+            Ok(Some(msg)) => {
+                eprintln!("{msg}");
             }
-        })
-        .filter(|x| !matches!(x.get_type(), Type::Nothing))
-        .into_pipeline_data(span, ctrlc)
-        .print_not_formatted(engine_state, false, true)?;
+            Err(err) => {
+                let working_set = StateWorkingSet::new(engine_state);
+                let msg = format_error(&working_set, &err);
+                eprintln!("{msg}");
+            }
```

Comment:
The old code did not exit on error, should we change this to exit early?

Reply:
I don't think so.  We can just report errors and try to remove other files.  Report and exit behavior seems buggy to me.